src/TesApi.Web/Management/IBatchQuotaVerifier.cs

```diff
@@ -12,17 +13,24 @@ namespace TesApi.Web.Management
     public interface IBatchQuotaVerifier
     {
         /// <summary>
-        /// Checks if the current quota allows fullfiment of the requested VM SKU.
+        /// Checks if the current quota allows fullfilment of the requested VM SKU.
         /// </summary>
-        /// <param name="virtualMachineInformation"></param>
+        /// <param name="virtualMachineInformation"></param>
+        /// <param name="needPoolQuotaCheck">A <see cref="Func{Boolean}"/> to enable checking pool quota.</param>
         /// <returns></returns>
         /// <exception cref="AzureBatchQuotaMaxedOutException">Thrown when a max quota condition was identified</exception>
-        Task CheckBatchAccountQuotasAsync(VirtualMachineInformation virtualMachineInformation);
+        Task CheckBatchAccountQuotasAsync(VirtualMachineInformation virtualMachineInformation, Func<bool> needPoolQuotaCheck);
```

Comment:
If we will be reusing an already active pool, the associated check isn't needed, I don't want that exception returned (too hard to separate it out from other legitimate exceptions) and I want all the remaining checks to also be performed (without otherwise changing the behavior of the method).

Reply:
If you want to add the new rule to the existing rules in some cases, or just use the existing rules, it would be better to implement the new rule as another method. This way, tests can be more specific and easier to implement, and it will be clearer what the caller's intent is.

Reply:
Almost the entire method would be identical (so a lot of duplicated code), and in my opinion it would be more confusing to build three methods to avoid this one parameter.

Reply:
I will replace the added parameter with a simple bool.